model_one/cli.py

```diff
@@ -1,198 +1,274 @@
-        magic_const = 32
-        if len(train_X) < magic_const or len(train_X) < magic_const or len(train_X) < magic_const or len(train_X) < magic_const:
-            raise Exception(
-                'Dataset must be contain at least {} elements'.format(magic_const))
-        return self._api_post_request('upload', build_data(self))
-
-    def upload(self, train_X, train_y, validate_X, validate_y):
-        from pandas import Series
-        if not (isinstance(train_X, Series) and isinstance(train_X, Series)
-                and isinstance(train_X, Series) and isinstance(train_X, Series)):
-            raise Exception('Try upload_arrays.')
-
-        return self.upload_arrays(train_X.tolist(), train_y.tolist(), validate_X.tolist(), validate_y.tolist())
-
-    def _inference(self, method, input):
-        if not input:
-            raise Exception('prompt must not be empty')
-
-        r = requests.get('{}/{}/{}?input={}'.format(_models_api(),
-                         self._id, method, urllib.parse.quote(input)), headers=_get_auth_header())
         if r.status_code != 200:
-            raise Exception(r.text)
+            raise BeyondmlModelException(r.text)
+
         return r.json()
 
-    def generate(self, input):
-        res = self._inference(method='generate', input=input)
-        return res['answer']['responses'][0]['response']
+    @classmethod
+    def create(cls, data: dict) -> dict:
+        return cls._request(*cls.VO["create"], data)
 
-    def classify(self, input):
-        res = self._inference(method='classify', input=input)
-        return res['answer']['scores']
+    @classmethod
+    def models(cls, data: dict) -> dict:
+        return cls._request(*cls.VO["models"], data)
 
-    def status(self):
-        if not self._id:
-            raise Exception('Initialize the model.')
-        r = requests.get('{}/{}/{}'.format(_models_api(),
-                         self._id, 'status'), headers=_get_auth_header())
-        if r.status_code != 200:
-            raise Exception(r.text)
-        self._status = r.json()['status']
-        return r.json()
+    @classmethod
+    def classify(cls, id: str, input: str) -> dict:
+        method, url = cls.V0["classify"]
+
+        return cls._request(method, url.format(id), params={"input": input})
+
+    @classmethod
+    def generate(cls, id: str, input: str) -> dict:
+        method, url = cls.V0["generate"]
+
+        return cls._request(method, url.format(id), params={"input": input})
+
+    @classmethod
+    def upload(cls, id: str) -> dict:
+        method, url = cls.V0["upload"]
 
+        return cls._request(method, url.format(id))
+
+    @classmethod
+    def status(cls, id: str) -> dict:
+        method, url = cls.V0["status"]
+
+        return cls._request(method, url.format(id))
+
+    @classmethod
+    def fit(cls, id: str) -> dict:
+        method, url = cls.V0["fit"]
+
+        return cls._request(method, url.format(id))
+
+
+class BeyondmlModelStatus(Enum, str):
```

Comment:
Enums are just convenient namespaces for constants, lets use them. They are also handy in case you need to change something as you wont need to search for the same unnamed string in different places

Reply:
And i've messed up parent classes ordering. `Enum` should come last